zeppelin-jupyter/src/main/java/org/apache/zeppelin/jupyter/JupyterUtil.java

```diff
@@ -0,0 +1,203 @@
+                (output instanceof ExecuteResult) ? ((ExecuteResult) output).getData()
+                    : ((DisplayData) output).getData();
+            for (Map.Entry<String, Object> datum : data.entrySet()) {
+              if (TEXT_PLAIN.equals(datum.getKey())) {
+                type = TypeData.TEXT;
+                result = Joiner.on(lineSeparator).join((List<String>) datum.getValue());
+              } else if (IMAGE_PNG.equals(datum.getKey())) {
+                type = TypeData.HTML;
+                result = makeHTML(((String) datum.getValue()).replace("\n", ""));
+              } else {
+                type = TypeData.TEXT;
+                result = datum.getValue().toString();
+              }
+              typeData = new TypeData(type, result);
+              typeDataList.add(typeData);
+            }
+          } else {
+            // Error
+            Error error = (Error) output;
+            type = TypeData.TEXT;
+            result = Joiner.on(lineSeparator)
+                .join(new String[]{error.getEname(), error.getEvalue()});
+            typeData = new TypeData(type, result);
+            typeDataList.add(typeData);
+          }
+        }
+      } else if (cell instanceof MarkdownCell) {
+        interpreterName = "%" + markdownReplaced;
+      } else {
+        interpreterName = "";
+      }
+
+      paragraph.setText(
+          interpreterName + lineSeparator + Joiner.on(lineSeparator).join(cell.getSource()));
+      paragraph.setResults(new Result(Result.SUCCESS, typeDataList));
+
+      paragraphs.add(paragraph);
+    }
+
+    note.setParagraphs(paragraphs);
+
+    return note;
+  }
+
+  private Gson getGson(GsonBuilder gsonBuilder) {
+    return gsonBuilder.registerTypeAdapterFactory(cellTypeFactory)
+        .registerTypeAdapterFactory(outputTypeFactory).create();
+  }
+
+  private String makeHTML(String image) {
+    return "<div style='width:auto;height:auto'><img src=data:image/png;base64," + image
+        + " style='width=auto;height:auto'/></div>";
+  }
+
+  public static void main(String[] args) throws ParseException, IOException {
+    Options options = new Options();
+    options.addOption("i", true, "Jupyter notebook file");
+    options.addOption("o", true, "Zeppelin note file. Default: note.json");
+
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd = parser.parse(options, args);
+
+    if (!cmd.hasOption("i")) {
+      new HelpFormatter().printHelp("java " + JupyterUtil.class.getName(), options);
+      System.exit(1);
+    }
+
+    Path jupyterPath = Paths.get(cmd.getOptionValue("i"));
+    Path zeppelinPath = Paths.get(cmd.hasOption("o") ? cmd.getOptionValue("o") : "note.json");
+
+    Note note = new JupyterUtil()
+        .getNote(new BufferedReader(new FileReader(jupyterPath.toFile())), "python", "md");
+
+    try (FileWriter fw = new FileWriter(zeppelinPath.toFile())) {
+      new GsonBuilder().setPrettyPrinting().create()
+          .toJson(note, fw);
```

Comment:
I have no preference but current Zeppelin's note format is multi-line json format. :-)